chore: update dependencies and refactor config plugin imports

Gauvino opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Switched to expo-background-task for more reliable background work and TV-safe behavior.
  - Added trickplay preview utilities and improved preview prefetching.
  - Downloads: added pause/resume/delete actions and improved modal/download start behavior.

- Bug Fixes
  - More robust Android manifest and PiP handling.
  - Chromecast/playback now validates missing API/user/item to prevent errors.
  - Improved notification cleanup.

- Refactor
  - Hooks/providers accept injected dependencies for clearer offline and API flows.

- Chores
  - Updated Expo packages, react-native-reanimated, scripts, and CI/typecheck; updated download-related translations.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Gauvino committed:
chore: update dependencies and refactor config plugin imports

- Removed deprecated dependency "@expo/config-plugins" and replaced with "expo/config-plugins" in multiple files.
- Updated various dependencies in package.json:
  - "expo" from "^53.0.6" to "^53.0.22"
  - "expo-router" from "~5.1.4" to "~5.1.5"
  - "expo-system-ui" from "~5.0.7" to "~5.0.11"
  - "react-native-reanimated" from "~3.16.7" to "~3.17.4"
- Cleaned up import statements for better compatibility with the latest Expo SDK.

Gauvino committed:
chore: add typecheck step to linting workflow and update build command

Gauvino committed:
fix: remove a warn and simplify notification cleanup logic in Layout component

Gauvino committed:
fix: update checkout ref handling and change runner version to ubuntu-24.04 in linting workflow

Gauvino committed:
Merge branch 'develop' into update-expo

Gauvino committed:
fix: resolve linting and safety issues across config files

Addresses TypeScript and ESLint warnings by adding void operator to async call without await

Adds missing expo-router babel plugin to ensure proper routing functionality

Improves Android manifest plugin safety with better null checks and error handling

Prevents duplicate network security config declarations and enhances error reporting with proper cause chaining

Gauvino committed:
fix: export correct function from Android manifest plugin

Corrects module export to use the actual implementation function instead of an undefined reference, preventing runtime errors when the plugin is imported and used.

fredrikburmester commented:
I can't build locally with this configuration. Any idea why? @Gauvino 

<img width="1280" height="333" alt="Screenshot 2025-08-29 at 13 40 39" src="https://github.com/user-attachments/assets/9ee94275-8ef1-4570-8e7e-2cbbbc8a8a0f" />

<img width="1280" height="327" alt="Screenshot 2025-08-29 at 13 40 27" src="https://github.com/user-attachments/assets/7e0f38a9-840c-49a0-be48-3ed1c0874521" />

Gauvino commented:
Ok i know why let me get on my pc and fix this

Gauvino committed:
feat: migrate from expo-background-fetch to expo-background-task

Replaces deprecated expo-background-fetch with the newer expo-background-task package to maintain background functionality. Updates all related imports, return values, and API calls throughout the codebase.

Improves error handling and logging in background task registration functions. Adds proper task registration checks to prevent duplicate registrations and enhances console output for better debugging.

Gauvino committed:
refactor: remove hook dependencies by accepting params

Decouples hooks from provider dependencies by accepting required data as parameters instead of importing hooks directly.

Improves testability and reduces coupling between components by making dependencies explicit through function parameters rather than internal hook calls.

Gauvino committed:
refactor: simplify background task registration for TV platforms

Consolidates TV platform handling by creating a conditional module reference instead of checking Platform.isTV in each function.

Removes TaskManager dependency and registration status checks to streamline the registration process.

Standardizes error logging format across all background task functions.

Gauvino committed:
fix: improve permission handling and background task reliability

Enhances notification permission flow by returning granted status and checking current permissions when previously asked. Prevents background tasks from registering when notifications are denied.

Improves background task registration with proper error handling, consistent intervals across platforms, and unregistration before re-registration to avoid conflicts.

Optimizes trickplay image prefetching with concurrency limiting to prevent I/O spikes and adds proper error handling.

Fixes type safety issues and improves error messaging in configuration plugins.

Gauvino committed:
Merge branch 'develop' into update-expo

Gauvino committed:
refactor: replace resizeMode with contentFit and improve error handling

Migrates image components from deprecated resizeMode prop to contentFit for better consistency across the codebase.

Adds comprehensive validation checks for API availability, user authentication, and item data before attempting stream operations to prevent runtime errors.

Extracts trickplay utilities into separate module for better code organization and reusability.

Enhances video player controls with additional API and download context parameters for improved functionality.

Gauvino committed:
feat: add pause/resume functionality to downloads

Enhances download management by allowing users to pause and resume active downloads instead of only being able to cancel them.

- Adds pause and resume buttons to download cards with appropriate visual states
- Implements error handling for pause/resume operations with user feedback
- Updates translations to reflect new pause/resume actions
- Refactors component from React.FC to default function export
- Removes unused imports and mutation logic in favor of direct async handlers

Gauvino committed:
refactor: improve background task registration with duplicate checks

Adds TaskManager import to check for already registered tasks before attempting registration, preventing unnecessary re-registration attempts.

Enhances error handling with console logging for better debugging while maintaining non-critical error behavior.

Fixes import statement to use default export for ActiveDownloads component.

Gauvino committed:
fix: improve modal dismissal timing and error handling

Ensures download modal is fully dismissed before initiating downloads to prevent UI conflicts.

Adds proper modal state handling with dismiss callback and keyboard interaction support.

Improves error logging by only showing console errors for unhandled resume failures.

fredrikburmester commented:
I get this warning @Gauvino 
```
| expo-router/babel is deprecated in favor of babel-preset-expo in SDK 50. To fix the issue, remove "expo-router/babel" from "plugins" in your babel.config.js file.
```

Gauvino commented:
> I get this warning @Gauvino 
> 
> ```
> 
> | expo-router/babel is deprecated in favor of babel-preset-expo in SDK 50. To fix the issue, remove "expo-router/babel" from "plugins" in your babel.config.js file.
> 
> ```

You're on a previous version I already remove this

fredrikburmester commented:
Looks great to me!

Gauvino commented:
Their is many change I Hope it's alright for everyone it my first time making such changes

fredrikburmester commented:
Can you list all changes you've made? @Gauvino

fredrikburmester commented:
<img width="377" height="168" alt="Screenshot 2025-08-29 at 21 54 41" src="https://github.com/user-attachments/assets/8abaf34b-19d8-44c7-a03b-c125f24f9891" />

i think this ui can be improved, maybe more padding or move the buttons, or something!

fredrikburmester commented:
<img width="126" height="148" alt="Screenshot 2025-08-29 at 21 55 12" src="https://github.com/user-attachments/assets/9297dcf9-5652-4b7c-9cb5-7a59feda6879" />
Yeah it's overlapping

Gauvino commented:
> Can you list all changes you've made? @Gauvino 

I can make a fullchangelog but not right now,

- Download can be paused and resume now
- Better error or warn logging for everything
- Fixed many warning
- Optimize trickplay and segments for performance
- Remove and replaced depracted package and code

fredrikburmester commented:
Awesome work. Pausing/resuming downloads works great for me

Gauvino commented:
@coderabbitai make me a changelog on everything in this PR

Gauvino commented:
Very good resume on what I did

fredrikburmester commented:
This looks great, and i've tested. I'll merge.

Gauvino commented:
> <img width="377" height="168" alt="Screenshot 2025-08-29 at 21 54 41" src="https://github.com/user-attachments/assets/8abaf34b-19d8-44c7-a03b-c125f24f9891" />
> 
> 
> 
> i think this ui can be improved, maybe more padding or move the buttons, or something! 

Yes i wanted to reduce a bit the size of the buttons but didn't have any more time

fredrikburmester commented:
That's fine. We can fix that small issue in develop.

Gauvino commented:
I Will do it when having a bit more time
